### PR TITLE
Bug 1223890 - Constraint errors in history panel

### PR DESCRIPTION
**Pull request by Johennes**

### CODE REVIEW

File: Client/Frontend/Widgets/SiteTableViewController.swift [hunk 1]
```diff
@@ -52,9 +52,13 @@ class SiteTableViewHeader : UITableViewHeaderFooterView {
             make.height.equalTo(0.5)
         }
 
+        // A table view will initialize the header with CGSizeZero before applying the actual size. Hence, the label's constraints
+        // must not impose a minimum width on the content view.
         titleLabel.snp_makeConstraints { make in
-            make.left.equalTo(contentView).offset(SiteTableViewControllerUX.HeaderTextMargin)
-            make.right.equalTo(contentView).offset(-SiteTableViewControllerUX.HeaderTextMargin)
+            make.left.equalTo(contentView).offset(SiteTableViewControllerUX.HeaderTextMargin).priority(999)
```
Comment (thebnich) [inline Client/Frontend/Widgets/SiteTableViewController.swift]: SnapKit includes a `priorityHigh()` -- I think that'd be slightly cleaner than manually defining a priority here.

File: Client/Frontend/Widgets/SiteTableViewController.swift [hunk 2]
```diff
@@ -52,9 +52,13 @@ class SiteTableViewHeader : UITableViewHeaderFooterView {
...
+            make.left.greaterThanOrEqualTo(contentView)
```
Comment (thebnich) [inline Client/Frontend/Widgets/SiteTableViewController.swift]: Are these `greaterThanOrEqualTo` and `lessThanOrEqualTo` lines necessary?

**Comment by Johennes** [comment]
I think that technically the `greaterThanOrEqualTo` and `lessThanOrEqualTo` constraints are required. When the view is initialized with a zero size, the left and right space constraints will break due to their non-required priority. That's what we want to not impose a minimum size on the view that would conflict with the initial zero size and result in the warning on the console. However, when these constraints break, it also means that the label's x position is no longer determined, unless we add any kind of fallback constraints.

So that's the theory. Practically, however, I could not see any issue on the console when omitting the `greaterThanOrEqualTo` and `lessThanOrEqualTo` fallback constraints. I thought that it'll still be safer to add them but if you think they're superfluous or throw up too many questions, I can remove them. Just let me know.

Other than that, I updated the PR and replaced the `999` literal with `priorityHigh`. I guess the `750` is still high enough for our purposes.

**Comment by Johennes** [comment]
@thebnich Anything else I should change? Shall I rather remove the `greaterThanOrEqualTo` / `lessThanOrEqualTo` fallback constraints.

**Comment by thebnich** [comment]
Works for me. Thanks!

Merged 889206b.

**Pull request closed without merging**